validation/superchain-version_test.go

```diff
@@ -268,7 +246,19 @@ func getBytecodeHash(ctx context.Context, chainID uint64, contractName string, t
 		return "", fmt.Errorf("%s: %w", addrToCheck, err)
 	}
 
-	return crypto.Keccak256Hash(code).Hex(), nil
+	// if the contract is known to have immutables, setup the filterer to mask the bytes which contain the variable's value
+	bytecodeImmutableFilterer, err := initBytecodeWithImmutables(code, contractName)
+	// the contract bytecode does not have immutables, use the deployed bytecode as-is to calculate hash
+	if err != nil {
+		return crypto.Keccak256Hash(code).Hex(), nil
+	}
```

Comment:
I'm not sure that using an error to communicate "does not have immutables" is the right pattern, since there are a couple of different errors that can be returned?

How about we either:
1. return a `hashImmutables` boolean as well as the code and the error, and use that in the control flow below to conditionally mask the bytecode
2. have the `.maskBytecode()` method work even if there are no immutable references (perhaps it operates on an empty slice) and just call it unconditionally (could rename it if we go this route).